nearcore/src/state_sync.rs

```diff
@@ -276,28 +296,20 @@ fn get_current_state(
     };
 
     if Some(&new_epoch_id) == was_last_epoch_done.as_ref() {
-        tracing::debug!(target: "state_sync_dump", ?shard_id, ?was_last_epoch_done, ?new_epoch_id, new_epoch_height, ?new_sync_hash, "latest epoch is done. No new epoch to dump. Idle");
-        Ok(StateDumpAction::Wait)
-    } else if epoch_manager.get_shard_layout(&prev_epoch_id)
-        != epoch_manager.get_shard_layout(&new_epoch_id)
+        return Ok(StateDumpAction::Wait);
+    }
+
+    let shard_layout = epoch_manager.get_shard_layout(&new_epoch_id)?;
+
+    if shard_layout.shard_ids().contains(shard_id)
+        && cares_about_shard(chain, shard_id, &new_sync_hash, &shard_tracker, &account_id)?
```

Comment:
For my education, why does it check cares_about_shard for an account_id? I thought each thread just dumps for one shard id and that's it.

Reply:
I think it's just so that the thread dumping this shard ID can tell whether the state should be there. I think most of the time this acount ID is None because nobody is running the state dumper on a validator, but it's `Some` in tests, where this `cares_about_shard()` will be the right way of telling whether this shard_id is going to be in the state snapshot